core/templates/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -71,28 +71,29 @@ import { SubtitledHtml } from 'domain/exploration/subtitled-html.model';
 
 
 export interface InteractionBackendDict {
-  'default_outcome': OutcomeBackendDict;
+  'default_outcome': OutcomeBackendDict | null;
```

Comment:
Add a comment explaining why this can be null

Reply:
In the codebase I could only find `default_outcome` and `solution` of Interaction object being set to `null` in testing related files.
Should a comment like this suffice?:
```
// A null default_outcome means that this interaction does not have a
// default outcome. Usually set to `null` while testing.
```
And for `solution`
```
// A null solution means that this interaction does not have a
// solution set. Usually set to `null` while testing.
```

Reply:
We should not add `null` as a type just for the sake of tests. Although it can be correct here that type is ` | null`  but try to find out a more better reason. Think in terms of the usage of this property and what null actually refers here. Maybe file `core/templates/state_domain.py` could help

Reply:
@mridul-netizen hey, I created a simple exploration and logged `default_outcome` and `solution`, there:
![image](https://user-images.githubusercontent.com/35102959/127216912-fc298fae-12f3-4edb-ac3e-0952f51f276c.png)
I saw that EndExploration had default_outcome as null, and solution was always initialised as `null` when creating a card. And is null until we add a hint and then a solution for it.
So I've added these comments:
![image](https://user-images.githubusercontent.com/35102959/127219333-afffaf2c-c9fb-4fe8-82e0-cbf040e09f36.png)
I've pushed these for now, tell me if there are any changes required.

Reply:
Yup looks good!